src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/semanticalAnalysis/binaryOperations/strategy/IdenticalOperandsStrategy.java

```diff
@@ -25,28 +22,14 @@
 final public class IdenticalOperandsStrategy {
     private static final String message = "Left and right operands are identical.";
 
-    private final static Set<IElementType> operationsToCheck = new HashSet<>();
-    static {
-        operationsToCheck.add(PhpTokenTypes.opEQUAL);
-        operationsToCheck.add(PhpTokenTypes.opIDENTICAL);
-        operationsToCheck.add(PhpTokenTypes.opNOT_EQUAL);
-        operationsToCheck.add(PhpTokenTypes.opNOT_IDENTICAL);
-        operationsToCheck.add(PhpTokenTypes.opGREATER);
-        operationsToCheck.add(PhpTokenTypes.opGREATER_OR_EQUAL);
-        operationsToCheck.add(PhpTokenTypes.opLESS);
-        operationsToCheck.add(PhpTokenTypes.opLESS_OR_EQUAL);
-        operationsToCheck.add(PhpTokenTypes.kwINSTANCEOF);
-    }
-
     public static boolean apply(@NotNull BinaryExpression expression, @NotNull ProblemsHolder holder) {
         final PsiElement left  = ExpressionSemanticUtil.getExpressionTroughParenthesis(expression.getLeftOperand());
         final PsiElement right = ExpressionSemanticUtil.getExpressionTroughParenthesis(expression.getRightOperand());
         if (null == left || null == right) {
             return false;
         }
 
-        final IElementType operation = expression.getOperationType();
-        if (operationsToCheck.contains(operation) && PsiEquivalenceUtil.areElementsEquivalent(left, right)) {
+        if (BinaryExpressionUtil.isComparison(expression) && PsiEquivalenceUtil.areElementsEquivalent(left, right)) {
```

Comment:
Please revert + create a separate ticket for internal milestone for migration - I'll do the same for pending refactoring topics this week.